D1371R3.md

```diff
@@ -385,29 +384,26 @@ Within the parentheses, `inspect` is equivalent to `switch` and `if` statements
 except that no conversion nor promotion takes place in evaluating the value
 of its condition.
 
-There are two clause variations types inside the `inspect` body:
-
-  - `=>` denotes expression yielding a value.
-  - `:` never yields a value, but instead denotes a statement that is executed
-    in the enclosing context. e.g, a `return` statement will return from the
-    function `inspect` is currently running in.
-
-`inspect` is an expression if it's encountered in an expression-only context,
-`trailing-return-type` is specified, or any `=>` clauses are present.
-It's a statement otherwise.
+`inspect` is an expression in all contexts. Depending on the enclosed patterns it
+may either yield a `void` result or a value, the type of which will be statically
+deduced from the patterns themselves or specified by a trailing return type. The
+deduction is analogous to that performed when determining the return type of a
+lambda expression. A pattern that passes control to a compound statement yields
+a `void` result. The return types of all patterns must match. If a trailing return
+type is provided, all patterns must result in an expression returning a type that
+is implicitly convertible to the trailing return type.
 
 When `inspect` is executed, its condition is evaluated and matched
 in order (first match semantics) against each pattern. If a pattern successfully
 matches the value of the condition and the boolean expression in the guard
 evaluates to `true` (or if there is no guard at all), control is passed to the
-statement or expression following the matched pattern label. If the guard
+expression or compound statement following the matched pattern label. If the guard
 expression evaluates to `false`, control flows to the subsequent pattern.
 
-If no pattern matches, none of the statements or expressions specified are executed.
-In that case, if `inspect` is a statment - control is passed to the next statement.
-If it's an expression, the behavor is undefined. Additionally, if a `:` clause of
-the `inspect` expression does not stop execution of the function, the behaviour is
-undefined.
+If no pattern matches, none of the expressions or compound statements specified
+are executed. In that case if the `inspect` expression yields `void`, control is
+passed to the next statement. If the `inspect` expression does not yield void,
+the behavior is undefined.
```

Comment:
Should be `std::terminate` in the non-`void` case.

Reply:
Done